### DISCUSSION TITLE
SMB file transfer

### DISCUSSION THREAD

**Discussion by kazepoy** [apache/hop#4733] [Ideas]
Is the addition of "PutSMB" and "GetSMB" components (actions) planned?

**Comment by bamaer** [comment]
what would those PutSMB and GetSMB (Samba?) actions do that is not yet possible over VFS?

**Reply by kazepoy** [reply]
it seems that SMB is not supported by [VFS ](https://hop.apache.org/manual/latest/vfs.html#_apache_hop_vfs_file_systems) in HOP, am I wrong?

**Reply by bamaer** [reply]
good point! you can access windows shares (up to some degree, not flawless), but probably not over the SMB protocol. feel free to create a feature request ticket so we can keep track of this

**Comment by kazepoy** [comment]
done, thx

**Comment by nadment** [comment]
It is not mentioned in Hop's documentation but seems to be supported by VFS.

Source: https://commons.apache.org/proper/commons-vfs/filesystems.html

CIFS

The CIFS (sandbox) filesystem provides access to a CIFS server, such as a Samba server, or a Windows share.

URI Format

smb://[ username[: password]@] hostname[: port][ absolute-path]

Examples

    smb://somehost/home

**Reply by kazepoy** [reply]
unfortunately, [vfs](https://github.com/apache/commons-vfs/blob/master/commons-vfs2-sandbox/pom.xml) is based on the abandoned [jcifs](https://www.jcifs.org) lib. this only supports deprecated smbv1
